__tests__/model-utils.test.js

```diff
@@ -30,3 +34,82 @@ describe("checkForeignPrimaryKey", () => {
+    const input = {};
+    const output = checkValidPostedComment(input);
+    expect(output).toBe(false);
+  });
+
+  test("Returns true if the postedComment only has keys: username, body, article_id", () => {
+    const input = { username: "hello", body: "world", article_id: 1 };
+    const output = checkValidPostedComment(input);
+    expect(output).toBe(true);
+  });
+
+  test("Returns false if the postedComment has any other key", () => {
+    const input = { username: "hello", body: "world", article_id: 1, extra: 2 };
+    const output = checkValidPostedComment(input);
+    expect(output).toBe(false);
+  });
+
+  test("Returns false is the postedComment is missing username or body", () => {
+    const input = { username: "hello", article_id: 1, extra: 2 };
+    const output = checkValidPostedComment(input);
+    expect(output).toBe(false);
+  });
+
+  test("Adds an article_id key if provided", () => {
+    const input = { username: "hello", body: "world" };
+    const article_id = 1;
+    const output = checkValidPostedComment(input, article_id);
+    expect(output).toBe(true);
+  });
+
+  test("Does not mutate the input", () => {
+    const input = { username: "hello", body: "world" };
+    const article_id = 1;
+    const inputCopy = JSON.parse(JSON.stringify(input))
+    checkValidPostedComment(input, article_id);
+    expect(input).toEqual(inputCopy);
+  })
+});
+
+describe("formatObjectToArray", () => {
+  test("Returns an outer array", () => {
+    const input = {};
+    const sortOrder = [];
+    const output = formatObjectToArray(input, sortOrder);
+    expect(output).toEqual([[]]);
+  });
+
+  test("Returns 1 [key, value] pair as a [value] inside an outer array", () => {
+    const input = { hello: "world" };
+    const sortOrder = ["hello"];
+    const output = formatObjectToArray(input, sortOrder);
+    expect(output).toEqual([["world"]]);
+  });
+
+  test("Returns all [key, value] pairs as arrays of [value1, value2, valueN] inside an outer array", () => {
+    const input = { hello: "world", timeIs: 14.46, dataType: ["array", true] };
+    const sortOrder = ["hello", "timeIs", "dataType"];
+    const output = formatObjectToArray(input, sortOrder);
+    expect(output).toEqual([["world", 14.46, ["array", true]]]);
+  });
+
+  test("Returns formatted nested array values in the key order specified", () => {
+    const input = { hello: "world", timeIs: 14.46, dataType: ["array", true] };
+    const sortOrder = ["timeIs", "dataType", "hello"];
+    const output = formatObjectToArray(input, sortOrder);
+    expect(output).toEqual([[14.46, ["array", true], "world"]]);
+  });
+
+  test("Does not mutate the input", () => {
+    const input = { hello: "world", timeIs: 14.46, dataType: ["array", true] };
+    const sortOrder = ["hello", "timeIs", "dataType"];
+    const inputCopy = JSON.parse(JSON.stringify(input));
+    formatObjectToArray(input, sortOrder);
+    expect(input).toEqual(inputCopy);
+  });
+});
```

Comment:
Really thorough testing, and you've stuck really stringently to the single purpose rule 🎉

Reply:
Thanks Niamh!